docs/logicapps-ai-course/04_agent_functionality/07-add-security-filters-for-agent-knowledge-trimming.md

```diff
@@ -0,0 +1,248 @@
+    1. Select the appropriate `Subscription` and `Azure AI Foundry/Hub Project`
+    1. Select your embedding model under `Model deployment`
+    1. Set `API key` as the Authentication type
+    1. Check the acknowledgement
+    1. Then click on `Next`
+1. On the next page, keep all the default values and click on `Next`.
+1. On the last page, rename your `Objects name prefix` to `hr-index`. Then press `Create`.
+
+### Step 2 - Add empty UserIds Field
+
+1. On the left navigation bar of your Azure AI Search instance, click on `Search management` and then click on `Indexes`.
+![Screenshot shows Indexes button.](media/07-add-security-filters-for-agent-knowledge-trimming/indexes.png)
+1. Click on `hr-index`.
+1. Click on `Fields` and then click `+ Add field`.
+![Screenshot shows + Add Field button.](media/07-add-security-filters-for-agent-knowledge-trimming/add-fields.png)
+1. On the pop-up on the right-hand, fill in the following information:  
+   - Field Name: `UserIds`
+   - Type: `Collection(Edm.String)`
+   - Check `Filterable`  
+   - Press `Save`
+![Screenshot shows + Add Field button.](media/07-add-security-filters-for-agent-knowledge-trimming/create-userids.png)
+
+### Step 3 - Obtain document chunk ids
+
+1. Start Visual Studio Code and select the `Extensions` tile.
+![Screenshot shows Extensions Tile on VSCode.](media/07-add-security-filters-for-agent-knowledge-trimming/extension.png)
+1. Search for the Rest Client and Select `Install`.
+![Screenshot shows REST Client Extension on VSCode.](media/07-add-security-filters-for-agent-knowledge-trimming/rest-client.png)
+1. Create a new file named `get-documents.rest`.
+1. Paste in the following code in your file.
+
+```text
+@endpoint = PUT-YOUR-SEARCH-SERVICE-ENDPOINT-HERE
+@index = hr-index
+@queryApiKey = PUT-YOUR-SEARCH-SERVICE-QUERY-API-KEY-HERE
+
+POST https://{{endpoint}}/indexes/{{index}}/docs/search?api-version=2025-09-01
+api-key: {{queryApiKey}}
+Content-Type: application/json
+    
+{
+   "search": "*",
+   "select": "chunk_id,title",
+   "top": 6,
+   "skip": 0
+}
+```
+You will need to replace the `endpoint` with your own. To do this,
+1. On Azure Portal, open your Azure AI Search instance. Copy the link next to `Url` and paste it in `get-documents.rest`. The `Url` will look something like this: `https://name-of-azure-search-service.search.windows.net`.
+
+You will need to replace the `queryApiKey` with your own. To do this,
+1. On Azure Portal, open your Azure AI Search instance. On the left navigation bar, click on `Settings` and then click on `Keys`.
+1. Under `Manage query keys`, copy the key and paste it in `get-documents.rest`.
+    ![Screenshot shows Azure AI Search Keys.](media/07-add-security-filters-for-agent-knowledge-trimming/azure-ai-search-keys.png)
+    
+Now press `Send Request`. You should have gotten a `value` array with the `@search.score`, `chunk_id` and `title`.
+
+### Step 4 - Populate User Ids for each document
```

Comment:
can we add a comment that this is just an example to show how to this and the users would want to corporate this into their ingestion pipeline